Tasks/Common/sdkutils.ts

```diff
@@ -143,26 +141,25 @@ export abstract class SdkUtils {
         // region into the service options. If credentials remain undefined, the sdk
         // will attempt to auto-infer from the host environment variables or, if EC2,
         // instance metadata
-        if (awsServiceOpts) {
-            if (!awsServiceOpts.credentials) {
-                const creds = await getCredentials(taskParams)
-                if (creds) {
-                    awsServiceOpts.credentials = await creds.getPromise()
-                }
-            }
-            if (!awsServiceOpts.region) {
-                awsServiceOpts.region = await getRegion()
+        if (!awsServiceOpts.credentials) {
```

Comment:
Any chance awsServiceOpts is undefined?

Reply:
If it is, we should fail the task. It is declared non nullable and is only called from defaultClients.ts. All of the calls there always construct a service opts, and it never makes any sense to do this call without them.